### PR TITLE
Makes anomalies and wormholes spawn harvested pierced realities

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Makes several anomalies and the wormhole vent spawn harvested pierced realities.

## Why It's Good For The Game
This introduces more false positives to the game. Heretics already leave plenty of clues to their presence, and harvested Pierced Realities being able to appear 'naturally' forces crew to investigate into them further rather than calling out their presence solely based on Harvested Realities existing.
(Same thing happened with the Xeno room on Corg, people starged to put in more effort instead of instantly calling out xenos)

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

image

</details>

## Changelog
:cl:
add: Made several Anomalies and Wormholes create harvested pierced realities upon expiring
code: Added a turf proc for generating harvested pierced realities
/:cl:

### CODE REVIEW

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -578,6 +578,25 @@ GLOBAL_LIST_EMPTY(created_baseturf_lists)
...
+			chosen_location = get_teleport_turf(src, precision) //Using the random teleportation logic here to find a destination turf
+		// We don't want them close to each other - at least 1 tile of seperation
+		var/list/nearby_things = range(1, chosen_location)
+		var/obj/effect/heretic_influence/what_if_i_have_one = locate() in nearby_things
+		var/obj/effect/visible_heretic_influence/what_if_i_had_one_but_its_used = locate() in nearby_things
+		if(what_if_i_have_one || what_if_i_had_one_but_its_used)
+			location_sanity++
+			continue
+		new /obj/effect/visible_heretic_influence(chosen_location)
+		spawned ++
```
Comment (itsmeow) [inline code/game/turfs/turf.dm]:
```suggestion
		spawned++
```

**Review by itsmeow — CHANGES_REQUESTED** [review]

**Review by EvilDragonfiend — DISMISSED** [review]
I think these will be spawned in open space or densified turf.
but also appearing time should be a bit desynced from the creation of anomalies.

For example, `addtime(generate_fake_pierced_realities(), rand(0, 300) SECONDS)`

File: code/game/turfs/turf.dm [hunk 2]
```diff
@@ -591,12 +591,14 @@ GLOBAL_LIST_EMPTY(created_baseturf_lists)
...
-		if(what_if_i_have_one || what_if_i_had_one_but_its_used)
+		if(what_if_i_have_one || what_if_i_had_one_but_its_used || isspaceturf(chosen_location))
 			location_sanity++
 			continue
-		new /obj/effect/visible_heretic_influence(chosen_location)
+		addtimer(CALLBACK(src, PROC_REF(create_new_fake_reality), chosen_location), rand(0, 500))
 		spawned ++
 
+/turf/proc/create_new_fake_reality(turf/F)
+	new new /obj/effect/visible_heretic_influence(F)
```
Comment (EvilDragonfiend) [inline code/game/turfs/turf.dm]: new new...?
also, please put a line.

Comment (CydiaLamiales) [reply code/game/turfs/turf.dm]: Already fixed it

File: code/game/turfs/turf.dm (hunk 1 above)
Comment (itsmeow) [reply code/game/turfs/turf.dm]: You did not resolve this bestie

**Review by itsmeow — APPROVED** [review]
Very good addition

**Pull request merged**